test/backend/range.in.cpp

```diff
@@ -81,3 +82,46 @@ NGRAPH_TEST(${BACKEND_NAME}, range)
         ASSERT_EQ(results, test.expected_result);
     }
 }
+
+NGRAPH_TEST(${BACKEND_NAME}, range_subgraph)
+{
+    // Create a graph for f(start,stop,step) = Range(start,stop,step).
+    auto start = make_shared<op::Parameter>(element::i32, Shape{});
+    auto stop = make_shared<op::Parameter>(element::i32, Shape{});
+    auto step = make_shared<op::Parameter>(element::i32, Shape{});
+    PartialShape out_max_shape{15};
+
+    auto range = make_shared<op::Range>(start, stop, step);
+    range->output(0).set_max_partial_shape(out_max_shape);
```

Comment:
I think we are likely to see the `set_max_partial_shape` happen after the graph is fully constructed (for example, from an annotation file) so let's move it down to the end to avoid any dependency on validate and infer shapes.